example-code/snippets/src/quantity/FormatsProvider.test.ts

```diff
@@ -0,0 +1,121 @@
+    const formatSpec = await FormatterSpec.create("TestSpec", format, unitsProvider, persistenceUnit);
+
+    const result = formatSpec.applyFormatting(50); // The persistence unit is meters, so this input value is 50 m.
+    // result in formatted value of 50 m
+    // __PUBLISH_EXTRACT_END__
+
+    assert.equal(result, "50.0 m");
+  });
+
+  it("SchemaFormatsProvider Formatting with Unit System provided", async () => {
+    // __PUBLISH_EXTRACT_START__ Quantity_Formatting.Schema_Formats_Provider_Simple_Formatting_With_Unit_System
+    const formatsProvider = new SchemaFormatsProvider(schemaContext, "metric");
+    const unitsProvider = new SchemaUnitProvider(schemaContext);
+    const persistenceUnit = await unitsProvider.findUnitByName("Units.M"); // or unitsProvider.findUnit("m");
+
+    formatsProvider.unitSystem = "imperial"; // This will cause the method to return the first presentation format for the KoQ that uses imperial units.
+    const formatProps = await formatsProvider.getFormat("AecUnits.LENGTH_LONG");
+    const format = await Format.createFromJSON("testFormat", unitsProvider, formatProps!);
+    const formatSpec = await FormatterSpec.create("TestSpec", format, unitsProvider, persistenceUnit);
+
+    const result = formatSpec.applyFormatting(50); // The persistence unit is meters, so this input value is 50 m.
+    // result in formatted value of 164'0 1/2"
+    // __PUBLISH_EXTRACT_END__
+
+    assert.equal(result, "164'0 1/2\"");
+  });
+
+  it("SchemaFormatsProvider Parsing", async () => {
+    // __PUBLISH_EXTRACT_START__ Quantity_Formatting.Schema_Formats_Provider_Simple_Parsing
+    const formatsProvider = new SchemaFormatsProvider(schemaContext, "metric");
+    const unitsProvider = new SchemaUnitProvider(schemaContext);
+    const persistenceUnit = await unitsProvider.findUnitByName("Units.M"); // or unitsProvider.findUnit("m");
+
+    const formatProps = await formatsProvider.getFormat("AecUnits.LENGTH_LONG");
+    const format = await Format.createFromJSON("testFormat", unitsProvider, formatProps!);
+    const parserSpec = await ParserSpec.create(format, unitsProvider, persistenceUnit);
+    const result = parserSpec.parseToQuantityValue("50 km");
+    // result.value 50000  (value in meters)
+    // __PUBLISH_EXTRACT_END__
+
+    assert.equal((result as ParsedQuantity).value, 50000);
+  });
+
+  it("adding a format", async () => {
+    // __PUBLISH_EXTRACT_START__ Quantity_Formatting.Mutable_Formats_Provider_Adding_A_Format
+    const formatsProvider = new ExampleFormatProvider();
+    const format: FormatDefinition = {
+      label: "NewFormat",
+      type: "Fractional",
+      precision: 8,
+      formatTraits: ["keepSingleZero", "showUnitLabel"],
+      uomSeparator: "",
+    };
+    await formatsProvider.addFormat("AecUnits.LENGTH", format); // Add a format with the name "AecUnits.LENGTH". The name could be anything as long as it is unique.
```

Comment:
Should we say that "the name could be anything", when one of our providers - `SchemaFormatsProvider` - throws if it receives a name in a format other than full-schema-item name?

Reply:
yeah... I've removed that section of the sentence, added a section for SchemaFormatsProvider in learning docs about throwing an error if not a ec full name when getting a format.